lib/runtime-core/src/lib.rs

```diff
@@ -117,8 +117,26 @@ pub fn validate(wasm: &[u8]) -> bool {
 
 /// The same as `validate` but with an Error message on failure
 pub fn validate_and_report_errors(wasm: &[u8]) -> ::std::result::Result<(), String> {
+    validate_and_report_errors_with_features(wasm, Default::default())
+}
+
+/// The same as `validate_and_report_errors` but with a Features.
+pub fn validate_and_report_errors_with_features(
+    wasm: &[u8],
+    features: backend::Features,
+) -> ::std::result::Result<(), String> {
     use wasmparser::WasmDecoder;
-    let mut parser = wasmparser::ValidatingParser::new(wasm, None);
+    let config = wasmparser::ValidatingParserConfig {
+        operator_config: wasmparser::OperatorValidatorConfig {
+            enable_simd: features.simd,
+            enable_bulk_memory: false,
+            enable_multi_value: false,
+            enable_reference_types: false,
+            enable_threads: false,
+        },
```

Comment:
I like this being explicit, so don't change it, but here's a neat Rust trick you can do:
```
wasmparser::OperatorValidatorConfig {
  enable_simd: features.simd,
  ..wasmparser::OperatorValidatorConfig::default()
}
```

This is useful especially when there are a lot of fields that you don't care about (this code assumes OperatorValidatorConfig is Default and gives `false` for those values (which is what `derive(Default)` will do))

Reply:
As it is the build will break when wasmparser changes their feature set. Do we want that?

Reply:
Hmm, I mean we'll be updating wasmparser manually, so I think it's fine either way.

If you think that's something you might want to see, then keep it, otherwise you can use the syntax above

Reply:
> error[E0599]: no function or associated item named `default` found for type `wasmparser::operators_validator::OperatorValidatorConfig` in the current scope